modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItEmbeddedClientMarshallingTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app.client;
+
+import static org.apache.ignite.catalog.definitions.ColumnDefinition.column;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.ignite.catalog.ColumnType;
+import org.apache.ignite.catalog.definitions.TableDefinition;
+import org.apache.ignite.compute.JobDescriptor;
+import org.apache.ignite.compute.JobTarget;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgMarshalingJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgumentAndResultMarshalingJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgumentStringMarshaller;
+import org.apache.ignite.internal.runner.app.client.Jobs.JsonMarshaller;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoArg;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoResult;
+import org.apache.ignite.internal.runner.app.client.Jobs.ResultStringUnMarshaller;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.table.Tuple;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for embedded client marshalers for Compute API.
```

Comment:
Not sure about "embedded client", what does it mean? We have "embedded API" and "client API", right?